Catalog Functionality

jkurz opened the pull request:
* catalog image when doing up
* add catalog function to separately catalog something
* do not fail if image exists in catalogit

jkurz commented:
closes #67

jritsema commented on cmd/harborAPI.go:
```diff
@@ -402,3 +403,40 @@ func DeleteShipmentEnvironment(username string, token string, shipment string, e
+	}
+
+	if Verbose && resp.StatusCode == 422 {
+		log.Println("contianer has already been cataloged. Do not fail, just pass gracefully.")
```
I would remove `"Do not fail, just pass gracefully"` from this message that the user sees.  Also, check spelling on `container`.

jritsema commented on cmd/harborAPI.go:
```diff
@@ -402,3 +403,40 @@ func DeleteShipmentEnvironment(username string, token string, shipment string, e
+	//build URI
+	values := make(map[string]interface{})
+	template, _ := uritemplates.Parse(catalogitURI + "/v1/containers")
+	uri, _ := template.Expand(values)
```
No need to use the uri template parser here since it's a static uri.

jritsema commented on cmd/catalog.go:
```diff
@@ -0,0 +1,103 @@
+
+	//iterate shipments
+	for shipmentName, shipment := range harborCompose.Shipments {
+		fmt.Printf("Starting %v ...\n", shipmentName)
```
Change "Starting" to "Cataloging".

jritsema commented on cmd/catalog.go:
```diff
@@ -0,0 +1,103 @@
+		}
+	}
+
+	fmt.Println("Containers were successfully cataloged.")
```
Should we consider saying, "Container `images` were cataloged" or "`Images` were cataloged" here?  

...Or even not say anything at all since this will be the new default and the user should assume it works unless it doesn't.

wilsonodk commented on cmd/harborAPI.go:
```diff
@@ -402,3 +403,35 @@ func DeleteShipmentEnvironment(username string, token string, shipment string, e
+	}
+
+	if Verbose && resp.StatusCode == 422 {
+		log.Println("contianer has already been cataloged.")
```
Misspelled

wilsonodk commented on cmd/harborAPI.go:
```diff
@@ -402,3 +403,35 @@ func DeleteShipmentEnvironment(username string, token string, shipment string, e
+		log.Println(string(body))
+	}
+
+	return "Successfully Cataloged Contianer " + container.Name, nil
```
Misspelled

jkurz commented:
* fixed all container misspellings. 
* updated logs to be more precise
* using same logic to pick containers to send to catalogit as all other harbor-compose functions

jritsema commented:
Thanks Josh!

The code is getting a bit repetitive.  Some refactoring is due soon.  Maybe after we add tests ;)
